### ISSUE TITLE
Core/Quest: Quest requirements

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#19425]
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

**Description:** Some quests like http://www.wowhead.com/quest=12070/rallying-the-troops are not proposed to some players

**Current behavior:** After completing the quest http://www.wowhead.com/quest=11982/raining-down-destruction the npc must offer the quest 12070, but for some players it didn't, It looks some requirements on core side are wrong.
I checked the DB and every thing looks correct.
If you try to do the hole quest chain with a new character you will not get any trouble.  Starting by https://woehead.way-of-elendil.fr/?quest=11981
When the issue happen once you complete 11982, instead getting a popup window to start 12070 you will get this http://i.imgur.com/JnVpmTI.jpg
In this case no error are displayed IG or into logs.
If you .add 12070  .rew 12070 .rem 12070 you still not able to get the quest: http://i.imgur.com/cN7OLSl.jpg

**Expected behaviour:** After rewarding 11982 the quest 12070  must be proposed

**Steps to reproduce the problem:**

1. The player that get this issue did almost all quests on grizzly Hills, So probably a miss understanding of a quest condition happen in this case
2. I tried to reproduce the issue but no clue
3. Some other private have the same issue with this quest: https://redmine.rising-gods.de/issues/23099

**Branch(es):** 3.3.5

**TC rev. hash/commit:** https://github.com/TrinityCore/TrinityCore/commit/5cb1555de6db43db03d6202b7edaa9c6835ecfab

**Comment by ariel-** [comment]
Ok, this is unrelated to quest changes made in 3b27a06265e4f022ec81b00ab99255246e8dd467

This time is quest_template logic:
```sql
SELECT ID, LogTitle FROM quest_template WHERE RewardNextQuest = 12070;
```

returns
```
    ID  LogTitle                  
------  --------------------------
 11982  Raining Down Destruction  
 12249  Ursoc, the Bear God       
```

Now, we're failing in Player::SatisfyQuestPrevChain. As it turns out, having either "Raining Down Destruction" and/or "Ursoc, the Bear God" in a status different than NONE (that is, not in quest log, not rewarded) prevents entirely "Rallying the troops" from ever showing up as available.

**Comment by ariel-** [comment]
RewardNextQuest (mentioned in wiki as NextQuestInChain on the examples, and as NextQuestIdChain on the explanations part) means:

> The quest entry from a creature or gameobject that ends a quest and starts a new one. The result is, that if you end the quest, the new quest instantly appears from the quest giver.

It shouldn't be used whatsoever to condition other quests, we already have _addon for that.

**Comment by ariel-** [comment]
Closed by 214b8e53c5598695915c1392702e059d79cfc260
